### PR TITLE
aggregate log export errors, ignore changed files

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes https://github.com/kubernetes-sigs/kind/issues/942

/cc @aojea 
I spent some time digging into this, and I think this is a more complete fix.

### CODE REVIEW

**Comment by BenTheElder** [comment]
/hold 
need to fix up some more errors handling... in other parts of the code

File: pkg/internal/cluster/logs/logs.go [hunk 1]
```diff
@@ -64,20 +64,30 @@ func Collect(nodes []nodes.Node, dir string) error {
...
 		// grab all logs under /var/log (pods and containers)
-		cmd := node.Command("tar", "--hard-dereference", "-C", "/var/log", "-chf", "-", ".")
+		cmd := node.Command(
+			"sh", "-c",
+			// Tar will exit 1 if a file changed during the archival.
+			// We don't care about this, so we're invoking it in a shell
+			// And masking out 1 as a return value.
+			// Fatal errors will return exit code 2.
+			// http://man7.org/linux/man-pages/man1/tar.1.html#RETURN_VALUE
+			`tar --hard-dereference -C /var/log -chf - . || (r=$?; [ $r -eq 1 ] || exit $r)`,
```
Comment (BenTheElder) [inline pkg/internal/cluster/logs/logs.go]: I would really prefer to not invoke shell, but this is not doing much and there actually doesn't seem to be another good solution.

there is no flag that disables this behavior, and we want to preserve other exit codes:
- from tar that are not equal to 1
- any exit code from `docker exec ...`

File: pkg/internal/cluster/logs/logs.go [hunk 2]
```diff
@@ -113,7 +123,8 @@ func Collect(nodes []nodes.Node, dir string) error {
 	}
 
 	// run and collect up all errors
-	return concurrent.Coalesce(fns...)
```
Comment (aojea) [inline pkg/internal/cluster/logs/logs.go]: ?

**Comment by aojea** [comment]
/lgtm

Thanks

**Pull request merged**